File: appinventor/components-ios/src/UltrasonicSensorUnit.swift
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright © 2025 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// https://www.apache.org/licenses/LICENSE-2.0

@objc public class UltrasonicSensorUnit: NSObject, OptionList {
  @objc public static let Centimeters = UltrasonicSensorUnit("cm", 0)
  @objc public static let Inches = UltrasonicSensorUnit("in", 1)

  private static let LOOKUP: [String: UltrasonicSensorUnit] = generateOptionsLookup(Centimeters, Inches)

  let value: String
  let intValue: Int32

  @objc private init(_ value: String, _ intValue: Int32) {
    self.value = value
    self.intValue = intValue
  }

  @objc public func toUnderlyingValue() -> AnyObject {
    return value as AnyObject
  }

  @objc public func toInt() -> Int32 {
    return intValue
  }

  @objc public static func fromUnderlyingValue(_ value: String) -> UltrasonicSensorUnit? {
    return LOOKUP[value]
  }
}

File: appinventor/components/src/com/google/appinventor/components/common/UltrasonicSensorUnit.java
// -*- mode: java; c-basic-offset: 2; -*-
// Copyright 2020 MIT, All rights reserved
// Released under the Apache License, Version 2.0
// http://www.apache.org/licenses/LICENSE-2.0

package com.google.appinventor.components.common;

import java.util.HashMap;
import java.util.Map;

/**
 * Defines a UltrasonicSensorUnit type used by the Lego Ev3 ultrasonic sensor.
 */
public enum UltrasonicSensorUnit implements OptionList<String> {
  Centimeters("cm", 0),
  Inches("inch", 1);

  private String value;
  private int intValue;

  UltrasonicSensorUnit(String mode, int intMode) {
    this.value = mode;
    this.intValue = intMode;
  }

  public String toUnderlyingValue() {
    return value;
  }

  public Integer toInt() {
    return this.intValue;
  }

  private static final Map<String, UltrasonicSensorUnit> lookup = new HashMap<>();

  static {
    for (UltrasonicSensorUnit mode : UltrasonicSensorUnit.values()) {
      lookup.put(mode.toUnderlyingValue(), mode);
    }
  }

  public static UltrasonicSensorUnit fromUnderlyingValue(String mode) {
    return lookup.get(mode);
  }
}

